lib/mongodb.js

```diff
@@ -1869,26 +1886,49 @@ MongoDB.prototype.ping = function(cb) {
   }
 };
 
+// Determine if a property must have an ObjectID value
+function shouldBeObjectID(propDef) {
+  if (!propDef) return false;
+  if (propDef.type === ObjectID || propDef.mongodb && propDef.mongodb.dataType === ObjectID) return true;
```

Comment:
I believe that `dataType` is usually going to be a string value (`{dataType: 'ObjectID'}`), not a class constructor. Unless you have code that's normalizing the dataType value from a string to a constructor? 

Please add tests to verify.